dev/dev_how_to_maintain_my_tests_and_examples.Rmd

```diff
@@ -0,0 +1,179 @@
+
+- The "FISHMAP_UPDATE_TEST_OUTPUTS" environment variable is used to update the outputs to be tested. If its value is "TRUE" then the outputs will be replaced and everything else will remain the same.
+
+We will therefore detail how this works for both models.
+
+### The "small" model
+
+By default, unit tests are run with this model. They read the results of the necessary functions and execute only the function to be tested. Here is a simplified example of a test implementation for the `fm_fit_model` function (you can find the full example in `tests/testthat/test-fm_fit_model.R`):
+
+```{r, eval = FALSE}
+# Testing for small model
+  
+test_resolution <- Sys.getenv("FISHMAP_TEST_RESOLUTION", unset = "small")
+
+  if(test_resolution == "small"){
+    
+    expected_outputs <- readr::read_rds(
+      system.file(
+        "examples",
+        paste0("part2_output_", test_resolution , ".rds"),
+        package = "FishMap")
+    )
+    
+    #' @description Testing that the result of `fm_fit_model` is stable
+    expect_equal(
+      object = expected_result,
+      expected = expected_outputs,
+      tolerance = 10e-6
+    )
+  }
+```
+
+Explanation of what the code does: 
+
+- It looks at which conditon we should run the test in, model "small" or "big".
+- It reads the outputs of the correct model
+- It compares the new or old results to make sure we have the same results as before the changes.
+
+Here is another example using the output of the `fm_load_data` function:
+
+```{r, eval=FALSE}
+if(test_resolution == "small"){
+  
+  fm_data_inputs <- readr::read_rds(system.file("examples", "part1_output_small.rds", package = "FishMap"))
+  
+}else if (test_resolution == "big") {
+  ## TODO use correct params for line below
+  # fm_data_inputs <- fm_load_data(k = 0.75, month_start = 10, month_end = 12, ...)
+}
+
+# run part2
+withr::with_seed(1234,{
+  
+  
+  fm_model_results <- fm_fit_model(fm_data_inputs = fm_data_inputs,
+                                   SE = 1,
+                                   data_source = 1,
+                                   data_obs = 2,
+                                   samp_process = 0,
+                                   b_constraint = 2,
+                                   cov_samp_process = 0,
+                                   biomass_temporal = 1,
+                                   sampling_temporal = 0,
+                                   lf_link = 0,
+                                   ref_data = "com",
+                                   EM = "est_b",
+                                   month_ref = 1)
+  
+})
+```
+
+Code explanation:
+
+- Since we are in "small", we read the saved results from the `fm_load_data` function
+- We run the model to be able to compare it later
+
+### The "big" model
+
+The unit tests for the "big" model are not yet implemented. Due to the amount of data, only some of it will need to be stored and in the `tests/testthat` folder as this will only be used for testing.
+
```

Comment:
like line 59-80, maybe add here an example chunk of a test file to show how the output are stored in `inst/` folder